app/services/github/configuration.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Github
+  # Note that all calls to the GitHub v3 API returns paginated results.
+  #
+  # @see https://developer.github.com/v3/#pagination
+  #
+  class Configuration
+    attr_reader :user
+
+    def initialize(user)
+      @user = user
+    end
+
+    # @see https://developer.github.com/v3/media/#request-specific-version
+    # @see https://developer.github.com/v3/#user-agent-required
+    #
+    # rubocop:disable Metrics/MethodLength
+    def set_options!
+      auth_criteria_present!
+
+      {
+        headers: {
+          'Accepts' => 'application/vnd.github.v3+json',
+          'User-Agent' => 'Oddball'
+        },
+        basic_auth: {
+          username: user.github_username,
+          password: user.personal_access_token
+        }
+      }
+    end
+    # rubocop:enable Metrics/MethodLength
+
+    # Many GitHub endpoints, including the Search Issues endpoint, can return
+    # up to 100 results per page.  In order to return the max, the
+    # per_page parameter must be passed.
+    #
+    # TODO: Maxing the page size at 100 is standing in place for implementing
```

Comment:
Maybe we can extract this out to an issue

Reply:
Good call

Reply:
Added in https://github.com/oddballio/magnifier/issues/50